src/js/actions/tools.js

```diff
@@ -793,6 +790,12 @@ define(function (require, exports) {
             toolStore = this.flux.store("tool"),
             tools = toolStore.getAllTools();
 
+        var _documentChangeHandler = function () {
+            this.transfer(resetBorderPolicies);
```

Comment:
You should never transfer inside a bare event handler. Instead, this should just call a flux action.